src/System.Slices/src/System/Span.cs

```diff
@@ -299,27 +299,17 @@ public bool Equals(Span<T> other)
                 return false;
             }
 
-            Enumerator thisIt = this.GetEnumerator();
-            Enumerator otherIt = other.GetEnumerator();
-            while (true)
+            var nonBoxingComparer = EqualityComparer<T>.Default;
```

Comment:
@krwq (cc: @adamsitnik), this is not blocking this commit, but could we optimize this even more by simply comparing memory? Walking every single byte in a Span<byte> (probably the main use of Span<T>) seems expensive in comparison.